pkg/resources/object_renaming_lists_and_sets.go

```diff
@@ -0,0 +1,518 @@
+	ObjectRenamingDatabaseInstance.ManuallyOrderedList = objectRenamingDatabaseManuallyOrderedListFromSchema(d.Get("manually_ordered_list").([]any))
+
+	d.SetId("identifier")
+
+	return ReadObjectRenamingListsAndSets(ctx, d, meta)
+}
+
+func UpdateObjectRenamingListsAndSets(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	if d.HasChange("list") {
+		// It wasn't working with d.getChange(). It was returning null elements in one of the test case's steps.
+		oldList := d.GetRawState().AsValueMap()["list"].AsValueSlice()
+		newList := d.GetRawConfig().AsValueMap()["list"].AsValueSlice()
+
+		oldListMapped := mapObjectRenamingDatabaseListItemFromValue(oldList)
+		newListMapped := mapObjectRenamingDatabaseListItemFromValue(newList)
+
+		addedItems, removedItems := ListDiff(oldListMapped, newListMapped)
+
+		for _, removedItem := range removedItems {
+			ObjectRenamingDatabaseInstance.List = slices.DeleteFunc(ObjectRenamingDatabaseInstance.List, func(item objectRenamingDatabaseListItem) bool {
+				return item == removedItem
+			})
+		}
+
+		ObjectRenamingDatabaseInstance.List = append(ObjectRenamingDatabaseInstance.List, addedItems...)
+	}
+
+	if d.HasChange("ordered_list") {
+		oldOrderedList := d.GetRawState().AsValueMap()["ordered_list"].AsValueSlice()
+		newOrderedList := d.GetRawConfig().AsValueMap()["ordered_list"].AsValueSlice()
+		oldOrderedListMapped := mapObjectRenamingDatabaseOrderedListItemFromValue(oldOrderedList)
+		newOrderedListMapped := mapObjectRenamingDatabaseOrderedListItemFromValue(newOrderedList)
+
+		itemsToAdd, itemsToRemove := ListDiff(oldOrderedListMapped, newOrderedListMapped)
+		for _, removedItem := range itemsToRemove {
+			ObjectRenamingDatabaseInstance.OrderedList = slices.DeleteFunc(ObjectRenamingDatabaseInstance.OrderedList, func(item objectRenamingDatabaseOrderedListItem) bool {
+				return item == removedItem
+			})
+		}
+
+		for _, addedItem := range itemsToAdd {
+			addedItem.Order = ""
+			ObjectRenamingDatabaseInstance.OrderedList = append(ObjectRenamingDatabaseInstance.OrderedList, addedItem)
+		}
+
+		// The implementation is not complete due to mentioned issues with computed order in DiffSuppressFunc
+	}
+
+	if d.HasChange("manually_ordered_list") {
+		_, newManuallyOrderedList := d.GetChange("manually_ordered_list")
+		hasNewItems := len(newManuallyOrderedList.([]any)) > len(ObjectRenamingDatabaseInstance.ManuallyOrderedList)
+
+		// Copy the value from external source and compare with the new state to create the final state that will be saved
+		finalState := ObjectRenamingDatabaseInstance.ManuallyOrderedList
+
+		for index, item := range ObjectRenamingDatabaseInstance.ManuallyOrderedList {
+			order := index + 1
+			newValue, err := collections.FindFirst(newManuallyOrderedList.([]any), func(item any) bool {
+				return item.(map[string]any)["order"].(int) == order
+			})
+
+			// Items were removed
+			if errors.Is(err, collections.ErrObjectNotFound) {
```

Comment:
I don't quite get the logic here: if we did not find the given order it can mean that only a single object was removed but all the subsequent ones exist, right?

Okay, I think I understand, so this is this assumption that there will be no gaps. But with this assumption, why we can't just initially check the new length and just trim initially?

Also, it would be nice to have this assumption validated on the resource level (i.e. that there are no gaps and that it starts with 1).